libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/ChoiceOptionsSet.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder.Dialogs.Choices;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Bot.Builder.Dialogs.Adaptive.Input
+{
+    /// <summary>
+    /// Sets the ChoiceFactoryOptions.
+    /// </summary>
+    public class ChoiceOptionsSet : ChoiceFactoryOptions, ITemplate<ChoiceFactoryOptions>
+    {
+        private readonly string template;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChoiceOptionsSet"/> class.
+        /// </summary>
+        public ChoiceOptionsSet()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ChoiceOptionsSet"/> class.
+        /// </summary>
+        /// <param name="obj">Choice options values.</param>
+        public ChoiceOptionsSet(object obj)
+        {
+            if (obj is string template)
+            {
+                this.template = template;
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<ChoiceFactoryOptions> BindAsync(DialogContext dialogContext, object data = null, CancellationToken cancellationToken = default)
+        {
+            if (template == null)
+            {
+                return this;
+            }
+
+            var languageGenerator = dialogContext.Services.Get<LanguageGenerator>() ?? throw new MissingMemberException(nameof(LanguageGeneration));
+            var lgResult = await languageGenerator.GenerateAsync(dialogContext, template, dialogContext.State).ConfigureAwait(false);
+            if (lgResult is ChoiceFactoryOptions cs)
+            {
+                return cs;
+            }
+            else if (lgResult is string str)
+            {
+                try
+                {
+                    var jObj = (JArray)JsonConvert.DeserializeObject(str);
```

Comment:
Are we sure a string will always be a JArray or we need some validations?

Reply:
This is how the Bind is made in the ChoiseSet class too. If the deserialization fails, it returns null.